### DISCUSSION TITLE
Full text search on the generated site.

### DISCUSSION THREAD

**Discussion by leira** [srid/neuron#272] [Ideas]
Currently, neuron generated site only supports search with title and meta data. Is there any plan to support fuzzy full text search? I think it is more important and useful than CLI fuzzy search, As research in the notes is probably done through the web interface.

**Comment by leira** [comment]
Did some random search found [tinysearch](https://endler.dev/2019/tinysearch/). It is a Rust based wasm implementation of Bloom Filter algorithm. It seems to have a good space and time performance. I didn't get into the details yet, maybe we can either use it directly, or seek some Haskell Bloom Filter wasm solution.

**Comment by NightMachinery** [comment]
Sourcegraph is good for websearching on one's own repo. I personally prefer CLI searching. `ugrep` seems promising in this regard, especially if the author polishes its interactive tooling.

**Comment by GenrihGrigoryan** [comment]
Here are some projects, that have full-text search:

https://mathieudutour.github.io/foam-gatsby-template/
https://github.com/cotes2020/jekyll-theme-chirpy
https://github.com/Maxence-L/notenote.link

**Comment by srid** [comment]
Full-text will easily become problematic for larger Zettelkastens. The browser will have to fetch megabytes of JSON index.

Metadata-only search (graph search) is a good compromise; see https://github.com/srid/neuron/issues/108#issuecomment-706484958
